File: Linear Search/LinearSearch.py
def linearSearch(array, size, target):
    """
    Performs a linear search on the array to find the target element.
    If found, prints the index of the target element; otherwise, indicates that the target is not found.
    """
    index = -1  # Initialize index to -1 indicating target is not found yet
    for i in range(size):
        if array[i] == target:
            index = i  # Update index if target is found
            break  # Exit the loop once the target is found
    if index != -1:
        print(f"Element matches in index: {index}")
    else:
        print(f"Not found the element {target} in the array")

def main():
    """
    Main function to take user input, create an array, and search for an element using linearSearch.
    """
    size = int(input("Enter the size of the array: "))
    print(f"Enter {size} elements in an array")
    arr = []
    for _ in range(size):
        arr.append(int(input()))
    element = int(input("Enter the element you want to search in the array: "))
    linearSearch(arr, size, element)

if __name__ == "__main__":
    main()

File: Linear Search/LinearSearch.cpp
#include<iostream>
using namespace std;

void linearSearch(int *array, int size, int target) {
    int index = -1;  // Initialize index to -1 indicating target not found
    for(int i = 0; i < size; i++) {
        if(array[i] == target) {
            index = i;  // Update index if target is found
            break;  // Exit loop once target is found
        }
    }
    if(index != -1) {
        cout << "Element matches in index: " << index << endl;
    } else {
        cout << "Not found the element " << target << " in the array" << endl;
    }
}

int main() {
    int size, element;
    cout << "Enter the size of the array: ";
    cin >> size;

    // Dynamically allocate array to allow runtime size determination
    int *arr = new int[size];
    cout << "Enter " << size << " elements in the array: ";
    for(int i = 0; i < size; i++) {
        cin >> arr[i];
    }

    cout << "Enter the element you want to search in the array: ";
    cin >> element;

    linearSearch(arr, size, element);

    delete [] arr;  // Free dynamically allocated memory
    return 0;
}